db/view/view_update_generator.cc

```diff
@@ -158,7 +158,8 @@ future<> view_update_generator::start() {
                             ::mutation_reader::forwarding::no);
 
                     inject_failure("view_update_generator_consume_staging_sstable");
-                    auto result = staging_sstable_reader.consume_in_thread(view_updating_consumer(s, std::move(permit), *t, sstables, _as, staging_sstable_reader_handle));
+                    auto result = staging_sstable_reader.consume_in_thread(view_updating_consumer(s, std::move(permit), *t, sstables, _as, staging_sstable_reader_handle),
+                        dht::tokens_filter(_db.get_keyspace_local_ranges(s->ks_name())));
```

Comment:
@bhalevy I don't think this solves the problem of staging sstables not being cleaned up. The situation under which staging sstables have to be cleaned up is when we execute two node ops in succession, with staging sstables still being left over from the first one. Example scenario:
* decomission node A
* streaming produces staging sstables
* view building starts from staging sstables <- token information is obtained here
* decomission node B

At this point, staging sstables produced in the first decomm contain non-owned tokens and have to be filtered. But this filtering happens with a token filter which was created after the first decomm and which is out-of-date by the time we start to the second decomm. Restarting the node solves this but it is not elegant.

Reply:
During the 2nd decommission, the new token mapping is still pending and the current one is valid throughout the node operation, so the problem is at the moment of applying it as the latest token_metadata / effective_replication_map.
Ideally, we should do that with a barrier so to wait until previous versions are no longer in use.
But that's out of scope for this PR.

Cc @avikivity

Reply:
But then I don't understand what this PR achieves. All staging sstables written as part of the current node-op are up-to-date w.r.t. to the current topology and don't require filtering. And now you say we don't need to filter for those left over from previous node ops. What is the use case here?

Reply:
Maybe after https://github.com/scylladb/scylladb/pull/11745 we it is safe to assume that any changes in the topology will be quickly picked up. But currently a single iteration can run for a long time and it is exactly at those times that cleanup is needed the most. I think that with https://github.com/scylladb/scylladb/pull/11745 we can keep this simple approach and don't need to subscribe to topology changes.

Reply:
The problem at hand is view updates after decommission was aborted.
In the particular case we were left we lots of large sstables in staging that were processed long after decommission was aborted and they were not cleaned up in any way.

Reply:
Ok, indeed this PR will already help that use-case.